### PR TITLE
Mason improvements

### PR DESCRIPTION
**Pull request by springmeyer**
This makes improvements to the `-DENABLE_MASON` support.

 - Brings in fixed expat package
 - Checks for `curl` which is currently a required dependency of mason. Follow https://github.com/mapbox/mason/issues/274 for news of when it may not be needed.
 - Avoids calling `find_package(Osmium ...)` since this has the unwanted side effect of bringing in system libs for expat and bzip2 (when we need to link ENABLE_MASON builds against mason-provded expat and bzip2): https://github.com/Project-OSRM/osrm-backend/pull/3212#issuecomment-257990620

# Issue

Targets #3234 and https://github.com/Project-OSRM/osrm-backend/pull/3212#issuecomment-257990620.

## Tasklist
 - [x] review
 - [x] adjust for comments

## Requirements / Relations

Fixing the expat package requires a change to mason. The `update_dependencies.sh` depends on github releases to update deps. So we have a chicken n egg situation were we can't update mason until tagged. And mason can't be tagged until this PR is proven to work. So I've updated mason directly. Not sure yet how this will play with subtree merge.

### CODE REVIEW

**Comment by springmeyer** [comment]
Working on travis. Ready for review. On travis their is one timeout failure that is unrelated to this PR.

**Comment by TheMarex** [comment]
> So we have a chicken n egg situation were we can't update mason until tagged.

I don't see how tagging a new mason version depends on this PR?

> So I've updated mason directly. Not sure yet how this will play with subtree merge.

Please don't do that. Not going through the merge script causes some real head-aches every time we try that again down the read. My suggestion would be to do a mason release, revert this local change and then use the script.

File: CMakeLists.txt [hunk 1]
```diff
@@ -400,6 +406,8 @@ if(ENABLE_MASON)
     set(ENV{CCACHE_CPP2} "true")
   endif()
 
+  include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include)
```
Comment (TheMarex) [inline CMakeLists.txt]: Can we add a note here that `find_package` is not compatible with `mason_use` hence we need to manually define the include path here?

**Comment by TheMarex** [comment]
@springmeyer is there anything that is blocking another mason release?

File: CMakeLists.txt (hunk 1 above)
Comment (springmeyer) [inline CMakeLists.txt]: Done in https://github.com/Project-OSRM/osrm-backend/commit/4751b8af3c8680f3bb35e4c0d09d4d5170edd397

**Comment by springmeyer** [comment]
> > So we have a chicken n egg situation were we can't update mason until tagged.

> I don't see how tagging a new mason version depends on this PR?

Sorry I was not very clear: My feeling is that it depends on this PR working. Now that it is working/passing, there is nothing blocking a mason release.

> > So I've updated mason directly. Not sure yet how this will play with subtree merge.

> Please don't do that. Not going through the merge script causes some real head-aches every time we try that again down the read. My suggestion would be to do a mason release, revert this local change and then use the script.

Will do 👍 

> @springmeyer is there anything that is blocking another mason release?

No. Will tag shortly and then update this PR.

**Comment by springmeyer** [comment]
Adjustments have been made after review. This is ready to merge once green.

**Comment by TheMarex** [comment]
Looks good to me. :+1:

**Pull request merged**